File: 3351-maximize-happiness-of-selected-children/maximize-happiness-of-selected-children.js
/**
 * @param {number[]} happiness
 * @param {number} k
 * @return {number}
 */
var maximumHappinessSum = function(happiness, k) {
    happiness.sort((a,b)=>a-b);
    let val=0,sum=0;
    for(let i=0;i<k;i++){
        var last=happiness[happiness.length-i-1];
        last-=val;
        if(last<=0){
            break;
        }
        sum+=last;
        val++;
    }
    return sum;
};

File: 3351-maximize-happiness-of-selected-children/maximize-happiness-of-selected-children.py
class Solution(object):
    def maximumHappinessSum(self, happiness, k):
        """
        :type happiness: List[int]
        :type k: int
        :rtype: int
        """
        sm=val=0
        happiness.sort()
        for i in range(k):
            last=happiness[len(happiness)-i-1]
            last-=val
            if last<=0: break
            sm+=last
            val+=1
        return sm